### PR TITLE
feat(associate_roles): add associate_role resource

### PR DESCRIPTION
**Pull request by VictorAvelar**
Enables associate roles management. See: [commercetools docs](https://docs.commercetools.com/api/projects/associate-roles#top)

### NEW FEATURES | UPGRADE NOTES | ENHANCEMENTS | BUG FIXES | EXPERIMENTS

- Adds the associate_role resource
- Adds associate_role resource example (`tf` file)

Depends on https://github.com/labd/commercetools-node-mock/pull/102

### CODE REVIEW

**Comment by demeyerthom** [comment]
Hi @VictorAvelar! 

I think the most common issue is that the mock is not up to data? If this is the case please take a look at this repo: https://github.com/labd/commercetools-node-mock. Easiest would be to copy one of the other resources and go from there (for example [category[(https://github.com/labd/commercetools-node-mock/blob/main/src/repositories/category.ts).

Note that you will need the following:
- a repository class to implement the required create and update logic
- wire the new repository in `src/repositories/index.ts`
- a service to configure the autowiring of most logic (see the same for [category](https://github.com/labd/commercetools-node-mock/blob/main/src/services/category.ts)
- wire the service in `src/services/index.ts`
With the above steps you should be able to have the mock handle a new resource

You can open a PR there and assign me if you want it reviewed!

And if this was **not** the problem you are having let me know what you are running in to ;)

File: internal/resources/associate_role/resource.go [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+	}
+}
+
+// Schema implements resource.Resource.
+func (*associateRoleResource) Schema(_ context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Associate Roles provide a way to group granular Permissions and assign " +
+			"them to Associates within a Business Unit.\n\n" +
+			"See also the [Associate Role API Documentation](https://docs.commercetools.com/api/projects/associate-roles)",
+		Version: 1,
```
Comment (demeyerthom) [inline internal/resources/associate_role/resource.go:160]: Can we remove this version field? Default terraform version is 0, so we can keep to that

Comment (demeyerthom) [reply internal/resources/associate_role/resource.go:160]: Preferably I would also have this method at the top, as it describes the actual schema the other methods use

File: internal/resources/associate_role/resource_test.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	key := "sales_manager_europe_region"
+	name := "Sales Manager - Europe"
+
+	config := testAssociateRoleConfig(id, name, key)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.TestAccPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		CheckDestroy:             testAssociateRoleDestroy,
+		Steps: []resource.TestStep{
```
Comment (demeyerthom) [inline internal/resources/associate_role/resource_test.go:26]: Can we add another test checking if all the updated fields actually get updated?

**Review by demeyerthom — COMMENTED** [review]
@VictorAvelar This looks great! I have some minor comments, mostly to maintain consistency etc.

File: internal/resources/associate_role/resource.go (hunk 1 above)
Comment (VictorAvelar) [reply internal/resources/associate_role/resource.go:160]: Understood, I was previously following the declaration order in the interface, but I'll certainly adhere to your request.

**Review by demeyerthom — APPROVED** [review]
LGTM

**Pull request merged**